### PR TITLE
Enumrable practice

### PR DESCRIPTION
**Pull request by Hamzaoutdoors**
# in this pull request we implement the following 😃 : 

- [x] Created modules to re-implement the methods all?, any?, and filter.
- [x] The modules were tested to check if they work correctly.
 
_Thanks in advance for your feedback 💚_

### CODE REVIEW

File: MyList.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require_relative 'enumerable'
+
+class MyList
+  include Enumerable
+  def initialize(int1, int2, int3, int4)
+    @list = [int1, int2, int3, int4]
+  end
+end
```
Comment (mikethreels) [inline MyList.rb]: - [x] great job on implementing the initialize method 👍  however according to the requirements MyList class should have the #initialize method that accepts *multiple arguments* not just 4. you can use the `*args` to specify arguments if you don't know the exact amount.

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module Enumerable
+  def my_all?
+    return unless block_given?
+
+    response = false
+    @list.each do |n|
+      response = true if yield(n)
+    end
```
Comment (mikethreels) [inline enumerable.rb:8]: - [ ] good job on the enumerables 👍 however according to the requirements the MyList class should implement the `#each` method that yields all members of @list in succession. this means that you should write that function yourself.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: MyList.rb (hunk 1 above)
Comment (Hamzaoutdoors) [reply MyList.rb]: Done, thank you for your feedback 👍🏼

File: MyList.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def initialize(*list)
+    @list = list
+  end
+end
+
+list = MyList.new(1, 2, 3, 4)
+
+puts 'Test #all?'
+puts(list.my_all? { |n| n < 5 })
+puts '|n| n < 5 Should return true'
```
Comment (rloterh) [inline MyList.rb:14]: - [x] OPTIONAL
This line is returning _false_ instead of _true_. Kindly endeavor to fix it.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Hamza & Azeez, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾 :shipit:.
Congratulations! 🎉🥂

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comment(s) marked **[OPTIONAL]** under this review:_
<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**